src/helper/mutations.ts

```diff
@@ -228,44 +159,108 @@ export function useCreateThreadMutation() {
 export function useCreateCommentMutation() {
   const auth = useAuth()
   const entityId = useEntityId()
-  const showToastNotice = useToastNotice()
+  const mutation = gql`
+    mutation createComment($input: ThreadCreateCommentInput!) {
+      thread {
+        createComment(input: $input) {
+          success
+        }
+      }
+    }
+  `
 
   const createCommentMutation = async function (
     input: ThreadCreateCommentInput
   ) {
-    const args = {
-      query: gql`
-        mutation createComment($input: ThreadCreateCommentInput!) {
-          thread {
-            createComment(input: $input) {
-              success
-            }
-          }
-        }
-      `,
-      variables: { input },
-    }
-
-    try {
-      const response = await authFetchThread(args, auth)
+    const success = await mutationFetch(auth, mutation, input)
 
-      if (response.thread.createComment?.success) {
-        return !!(await mutate(`comments::${entityId}`))
-      }
-    } catch (e) {
-      handleError(showToastNotice, e)
-      return false
-    }
+    if (success) await mutate(`comments::${entityId}`)
+    return success
   }
 
   return async (input: ThreadCreateCommentInput) =>
     await createCommentMutation(input)
 }
 
-function handleError(
-  showToastNotice: ReturnType<typeof useToastNotice>,
-  response?: object
-) {
-  console.log(response)
-  showToastNotice(`Das hat leider nicht geklappt…`, 'warning')
+type MutationInput =
+  | NotificationSetStateInput
+  | UuidSetStateInput
+  | ThreadCreateThreadInput
+  | ThreadSetThreadArchivedInput
+  | ThreadSetThreadStateInput
+  | ThreadSetCommentStateInput
+  | ThreadCreateCommentInput
+
+type ApiErrorType = 'UNAUTHENTICATED' | 'FORBIDDEN'
+type ErrorType = ApiErrorType | 'UNKNOWN'
+
+export interface ApiError extends GraphQLError {
+  extensions: {
+    code: ApiErrorType
+  }
+}
+
+export async function mutationFetch(
```

Comment:
@inyono its is the relevant function (that I use in place of `createAuthAwareGraphqlFetch` atm.)

Reply:
(https://github.com/serlo/frontend/blob/937faf1c80f1decf1f719b6ed3313ae5325d5d0c/src/helper/mutations.ts#L203)